src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -337,169 +278,61 @@ void ProcessSelectedMethods(XPathNavigator nav, TypeDefinition type, object? cus
             }
         }
 
-        protected virtual void ProcessMethod(TypeDefinition type, XPathNavigator nav, object? customData)
+        protected virtual void ProcessMethod(TypeDesc type, XPathNavigator nav, object? customData)
         {
             string signature = GetSignature(nav);
             if (!String.IsNullOrEmpty(signature))
             {
-                MethodDefinition? method = GetMethod(type, signature);
-                if (method == null)
-                {
-                    LogWarning(nav, DiagnosticId.XmlCouldNotFindMethodOnType, signature, type.GetDisplayName());
-                    return;
-                }
-
-                ProcessMethod(type, method, nav, customData);
+                foreach (MethodDesc meth in type.GetMethods())
+                    if (signature == GetMethodSignature(meth, false))
+                        ProcessMethod(type, meth, nav, customData);
             }
 
             string name = GetAttribute(nav, NameAttributeName);
             if (!String.IsNullOrEmpty(name))
             {
-                bool foundMatch = false;
-                if (type.HasMethods)
+                foreach (MethodDesc method in type.GetAllMethods())
                 {
-                    foreach (MethodDefinition method in type.Methods)
+                    if (name == method.Name)
                     {
-                        if (name == method.Name)
-                        {
-                            foundMatch = true;
-                            ProcessMethod(type, method, nav, customData);
-                        }
+                        ProcessMethod(type, method, nav, customData);
                     }
                 }
-
-                if (!foundMatch)
-                {
-                    LogWarning(nav, DiagnosticId.XmlCouldNotFindMethodOnType, name, type.GetDisplayName());
-                }
             }
         }
 
-        protected virtual MethodDefinition? GetMethod(TypeDefinition type, string signature) => null;
-
-        protected virtual void ProcessMethod(TypeDefinition type, MethodDefinition method, XPathNavigator nav, object? customData) { }
+        protected virtual MethodDesc? GetMethod(TypeDesc type, string signature) => null;
 
-        void ProcessSelectedEvents(XPathNavigator nav, TypeDefinition type, object? customData)
```

Comment:
Why delete these?

Github's differ gets pretty confused at this point.

Reply:
That's a good question I know that NativeAOT uses the tables just like the ILtrim project, but there is no notion of EventDesc or PropertyDesc. My guess is that everything is represented as a method(?). Looking at the current implementation a type seems to only be able to have attributes and methods

Reply:
It might even affect the schema PR https://github.com/dotnet/linker/pull/2500

Reply:
There is a PropertyPseudoDesc if really needed but this can be ifdeffed out for now.